File: midterm lab 3/java/MP09.java
import java.io.*;
import java.util.*;

// MP09 - Display dataset statistics
public class MP09 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        try {
            System.out.print("Enter CSV file path: ");
            String filePath = scanner.nextLine();

            BufferedReader br = new BufferedReader(new FileReader(filePath));

            String line;
            int rowCount = 0;
            int columnCount = 0;

            while ((line = br.readLine()) != null) {
                String[] data = line.split(",");
                
                // Count columns (only once)
                if (rowCount == 0) {
                    columnCount = data.length;
                }

                rowCount++;
            }

            br.close();

            // Display stats
            System.out.println("\nDataset Statistics:");
            System.out.println("Total Rows: " + rowCount);
            System.out.println("Total Columns: " + columnCount);

        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

File: midterm lab 3/javascript/MP09.js
// MP09 - Display dataset statistics

const fs = require("fs");
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter CSV file path: ", (filePath) => {
    try {
        const data = fs.readFileSync(filePath, "utf-8");

        const lines = data.trim().split("\n");

        const rowCount = lines.length;
        const columnCount = lines[0].split(",").length;

        console.log("\nDataset Statistics:");
        console.log("Total Rows:", rowCount);
        console.log("Total Columns:", columnCount);

    } catch (err) {
        console.log("Error:", err.message);
    }

    rl.close();
});